course02/project01/gavin/Asteroids/collidable.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "game_object.h"
+
+
+class Collidable : public GameObject {
+public:
+    Collidable(GameWorld* root, ObjectId id, glm::vec2 position, float size);
```

Comment:
I think I should use some kind of type trait on my ```GameObject```s now to demonstrate that they are Collidable, and have a ```Collide``` method. On the other hand I still like having this abstract class so I don't have to re-implement the ```Overlap``` behavior for every type.